File: src/code_generation/output.js
class MathUtils {
    let count : Number = 0
    let name : string = 2
    let base : Number

    function factorial(self, n) : Number {
        if (n > 0) {
            return n * self.factorial(n - 1)
        }
        else {
            return 1
        }
    }

File: src/code_generation/output.java
public class MathUtils {
    private int count = 0;
    private string name = 2;
    private int base;
    public int factorial(Object self, Object n) {
        if (n > 0) {
            return n * self.factorial(n - 1);
        } else {
            return 1;
        }
    }
}